File: diary/May27/MoveZeroes.java
/**
 * Input: [0,1,0,3,12] Output: [1,3,12,0,0]
 */

class MoveZeroes {
    static void moveZeroes(int[] nums) {
        if (nums.length == 0) {
            return;
        }
        int j = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != 0) {
                nums[j] = nums[i];
                j++;
            }
        }
        for (int i = j; i < nums.length; i++) {
            nums[i] = 0;
        }

        for (int i = 0; i < nums.length; i++) {
            System.out.println(nums[i]);
        }
    }

    public static void main(String[] args) {
        int[] nums = new int[] { 0, 1, 0, 3, 12 };
        moveZeroes(nums);
    }
}

File: diary/May27/moveZeroes.js
/**
 * 
 * @param {*} nums [0,1,0,3,12]
 * [1,3,12,0,0]
 */
var moveZeroes = function (nums) {
    let count = 0, pos;
    //when the item of 0 is found, the position
    while (~(pos = nums.indexOf(0))) {
        count++;
        //remove 1 item at index pos, return the rest
        nums.splice(pos, 1);
    }
 nums.splice(nums.length, 0, ...((new Array(count)).fill(0)));
 return nums;
};
console.log(moveZeroes([0,1,0,3,12]));